src/app/features/courses/courses.component.html

```diff
@@ -1,13 +1,25 @@
 <div class="courses">
   <div class="container">
-    <app-courses-search></app-courses-search>
+    <app-courses-search
+      [(searchValue)]="searchValue"
```

Comment:
This is redundant. You don't need two-way data binding here, since your search logic implementation must be encapsulated in the app-courses-search component. You communicate with parents via `@Output`.